src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -973,16 +970,35 @@ public function getDirty()
     }
 
     /**
-     * Determine if the new and old values for a given key are numerically equivalent.
+     * Determine if the new and old values for a given key are equivalent.
      *
-     * @param  string  $key
+     * @param  string $key
+     * @param  mixed  $current
      * @return bool
      */
-    protected function originalIsNumericallyEquivalent($key)
+    protected function originalIsEquivalent($key, $current)
     {
-        $current = $this->attributes[$key];
+        if (! array_key_exists($key, $this->original)) {
+            return false;
+        }
 
-        $original = $this->original[$key];
+        if ($current === $original = $this->getOriginal($key)) {
+            return true;
+        }
+
+        // When check rich this check and current attribute value not equals with original, we should skip next steps
+        // if current is null
+        if (is_null($current)) {
```

Comment:
This check for nullable may be should move to [fromDateTime](https://github.com/laravel/framework/blob/5.4/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L730) method as like [castAttribute](https://github.com/laravel/framework/blob/5.4/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L458) method.

This hack check looked from [setAttribute](https://github.com/laravel/framework/blob/5.4/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L525)

Reply:
The missing `is_null($original)` causes problems for `deleted_at` which is NULL before a model is updated.